### PR TITLE
Add opcodes for `tx.*`

### PR DESCRIPTION
**Pull request by 0xRampey**
Probably best to merge #8 or #9 before getting this in

### CODE REVIEW

File: erc20/src/lib.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ impl ERC20 {
         if owner != address!("0000000000000000000000000000000000000007") {
             revert();
         }
+        if tx_gas_price() != U256::from(42) {
+            revert();
+        }
```
Comment (leonardoalt) [inline erc20/src/lib.rs:41]: Why this? Just to use it as an example?

File: eth-riscv-syscalls/src/lib.rs [hunk 2]
```diff
@@ -60,4 +62,6 @@ syscalls!(
     (4, Revert, "revert"),
     (5, Caller, "caller"),
     (0x20, Keccak256, "keccak256"),
+    (0x3A, GasPrice, "gasprice"),
+    (0x32, Origin, "origin"),
```
Comment (leonardoalt) [inline eth-riscv-syscalls/src/lib.rs:66]:
```suggestion
    (0x32, Origin, "origin"),
    (0x3A, GasPrice, "gasprice"),
```
just to keep sorted by opcode

File: eth-riscv-syscalls/src/lib.rs [hunk 3]
```diff
@@ -52,6 +52,8 @@ macro_rules! syscalls {
 // t0: 4, opcode for revert, doesn't return
 // t0: 5, opcode for caller, returns an address
 // t0: 0x20, opcode for keccak256, a0: offset, a1: size, returns keccak256 hash
+// t0: 0x3A, opcode for gasprice, returns 256-bit value
+// t0: 0x32, opcode for origin, returns an address
```
Comment (leonardoalt) [inline eth-riscv-syscalls/src/lib.rs:56]: here too

File: erc20/src/lib.rs (hunk 1 above)
Comment (0xRampey) [reply erc20/src/lib.rs:41]: Yeah, do we want to keep it for unit tests instead?

Comment (leonardoalt) [reply erc20/src/lib.rs:42]: yea I think it'd be better to keep this example as clean as possible

Comment (0xRampey) [reply erc20/src/lib.rs:42]: Makes sense!

**Review by gakonst — APPROVED** [review]

**Pull request merged**